src/spiral_stc.cpp

```diff
@@ -8,294 +8,356 @@
+        // turn ccw
+        dx = pathNodes.back().pos.x - prev.pos.x;
+        dy = pathNodes.back().pos.y - prev.pos.y;
+        dx_prev = dx;
+        dx = -dy;
+        dy = dx_prev;
+      }
+      else
       {
-        if (grid[y2][x2] == eNodeOpen && visited[y2][x2] == eNodeOpen)
+        // Initialize spiral direction towards y-axis
+        dx = 0;
+        dy = 1;
+      }
+      done = true;
+
+      for (int i = 0; i < 4; ++i)
+      {
+        x2 = pathNodes.back().pos.x + dx;
+        y2 = pathNodes.back().pos.y + dy;
+        if (x2 >= 0 && x2 < nCols && y2 >= 0 && y2 < nRows)
         {
-          Point_t new_point = { x2, y2 };
-          gridNode_t new_node =
+          if (grid[y2][x2] == eNodeOpen && visited[y2][x2] == eNodeOpen)
           {
-            new_point,  // Point: x,y
-            0,          // Cost
-            0,          // Heuristic
-          };
-          prev = pathNodes.back();
-          pathNodes.push_back(new_node);
-          it = --(pathNodes.end());
-          visited[y2][x2] = eNodeVisited;  // Close node
-          done = false;
-          break;
+            Point_t new_point = {x2, y2};
+            gridNode_t new_node =
+                {
+                    new_point, // Point: x,y
+                    0,         // Cost
+                    0,         // Heuristic
+                };
+            prev = pathNodes.back();
+            pathNodes.push_back(new_node);
+            it = --(pathNodes.end());
+            visited[y2][x2] = eNodeVisited; // Close node
+            done = false;
+            break;
+          }
         }
+        // try next direction cw
+        dx_prev = dx;
+        dx = dy;
+        dy = -dx_prev;
       }
-      // try next direction cw
-      dx_prev = dx;
-      dx = dy;
-      dy = -dx_prev;
     }
+    return pathNodes;
   }
-  return pathNodes;
-}
 
-std::list<Point_t> SpiralSTC::spiral_stc(std::vector<std::vector<bool> > const& grid,
-                                          Point_t& init,
-                                          int &multiple_pass_counter,
-                                          int &visited_counter)
-{
-  int x, y;
-  // Initial node is initially set as visited so it does not count
-  multiple_pass_counter = 0;
-  visited_counter = 0;
-
-  std::vector<std::vector<bool> > visited;
-  visited = grid;  // Copy grid matrix
-  x = init.x;
-  y = init.y;
-
-  Point_t new_point = { x, y };
-  gridNode_t new_node =
+  std::list<Point_t> SpiralSTC::spiral_stc(std::vector<std::vector<bool>> const &grid,
+                                           Point_t &init,
+                                           int &multiple_pass_counter,
+                                           int &visited_counter)
   {
-    new_point,  // Point: x,y
-    0,          // Cost
-    0,          // Heuristic
-  };
-  std::list<gridNode_t> pathNodes;
-  std::list<Point_t> fullPath;
-  pathNodes.push_back(new_node);
-  visited[y][x] = eNodeVisited;
+    int x, y;
```

Comment:
You're not using these until assignment here: `x = init.x;`
So just declare them there.